### ISSUE TITLE
Tabbing/Navigation over all kind of items using the Tab key

### ISSUE THREAD

**Issue by JPGygax68** [ocornut/imgui#3092]
I cannot seem to navigate between my text field and my tree view (tree of TreeNode widgets) using the Tab key.

I can get to the text field (which is first child of my window) using Tab.
When I click on my text field, I can tab out of it, but the focus does not go to my tree.
When I click on any node of my tree, I can use the cursor keys to navigate and open/close nodes, but I cannot tab back to my text field.

The tree is wrapped into a `BeginChild`/`EndChild` group; the `InputText` widget is not.

Note: using cursor up/down, I can get a frame to display around either my text field or my tree; however, that seems to have no influence on the keyboard focus.

**Comment by ocornut** [comment]
Hello Hans-Peter,

Essentially none of the things you are (justly) suggesting are presently implemented.
The Tabbing system was implemented very early on (<1.00) and the keyboard/gamepad navigation system doesn't really uses it.

I had a large wip stash working on redesigning this but the feature set kinda grew (as I was working on an API to activate/focus item which requires solving more issues). I could perhaps try to get a simpler version of that in.

**Comment by JPGygax68** [comment]
Thanks Omar! Whatever you can do will be welcome. I quite understand that focus management is far from trivial.

**Comment by slajerek** [comment]
Thanks for mentioning my issue. This is very serious blocker for me as my tool needs quick navigation using keyboard, that is quite a main requirement. Good luck on getting this done!

**Comment by ocornut** [comment]
- [ ] Another thing we will need take into account is Tabbing through `InputTextMultiline()` widgets. When they are configured to allow tab inputs, they cannot be tabbed out, and pressing ESC followed by Tab refocus the same item. This needs to be fixed as well.

**Comment by ocornut** [comment]
I have pushed the remaining glue to allow Tabbing to cycle through every items.

Note that "full" tabbing requires `ImGuiConfigFlags_NavEnableKeyboard` to be enabled.
Without `ImGuiConfigFlags_NavEnableKeyboard`, tabbing will imitate old behavior which is to only cycle through text-inputable inputables.

I have also switched examples application to enable both ImGuiConfigFlags_NavEnableKeyboard and ImGuiConfigFlags_NavEnableGamepad by default now.
